weasyl/oauth2.py

```diff
@@ -6,12 +6,20 @@
 from libweasyl.oauth import get_consumers_for_user, revoke_consumers_for_user, server
 from weasyl.controllers.decorators import disallow_api, token_checked
 from weasyl import define as d
-from weasyl import errorcode, http, login, media, orm
+from weasyl import errorcode, login, media, orm
+
+
+class OAuthResponse(Response):
+    def __init__(self, headers, body, status):
+        super(OAuthResponse, self).__init__(
+            body=body,
+            status_code=status,
+            headers={k.encode('utf-8'): v.encode('utf-8') for k, v in headers.iteritems()},
+        )
 
 
 def extract_params(request):
-    headers = http.get_headers(request.environ)
-    return request.path, request.method, request.params, headers
+    return request.path, request.method, request.params, request.headers
```

Comment:
Should https://github.com/Weasyl/weasyl/blob/9dea21db3c8ddb22dbf804366834df9d2feb7f5a/weasyl/middleware.py#L350 change as well?